modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptFilter.kt

```diff
@@ -38,6 +40,30 @@ class TranscriptRegexFilters(private val filters: List<TranscriptFilter>) : Tran
                 RegexFilters.tripleOrMoreEmptyLinesFilter,
             ),
         )
+
+        fun extractSpeaker(cue: String, format: TranscriptFormat?) = when (format) {
+            TranscriptFormat.VTT -> regexMatch(input = cue, pattern = "<v (.+?)>", position = 1)
+            TranscriptFormat.SRT -> regexMatch(input = cue, pattern = "^(.+?):", position = 1)
+            else -> null
+        }
```

Comment:
I could be overthinking this, but is it bad / slow to be creating the regex pattern every time? How about something like the following:
```kotlin
        private val vttRegex = "<v (.+?)>".toRegex()
        private val srtRegex = "^(.+?):".toRegex()

        fun extractSpeaker(cue: String, format: TranscriptFormat?) = when (format) {
            TranscriptFormat.VTT -> regexMatch(input = cue, regex = vttRegex, position = 1)
            TranscriptFormat.SRT -> regexMatch(input = cue, regex = srtRegex, position = 1)
            else -> null
        }

        private fun regexMatch(input: String, regex: Regex, position: Int = 0): String? {
            return try {
                regex.find(input)?.groupValues?.get(position)
            } catch (e: Exception) {
                null
            }
        }
```

Reply:
I extracted all regex patterns outside in 109b44f and 8a7721c, hopefully it improves performance even more!